From 5feb66e855d26dc2f08b323e24a50b3ee25f2adf Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:46 -0400
Subject: [usb] core: allow isoc URBs for wireless devices with an interval < 6

Message-id: <1410887463-20674-215-git-send-email-dzickus@redhat.com>
Patchwork-id: 95173
O-Subject: [RHEL7 PATCH 214/591] usb: core: allow isoc URBs for wireless devices with an interval < 6
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit a2cd612dc9be768b516541b2164c843849bc3bb3
Author: Thomas Pugliese <thomas.pugliese@gmail.com>
Date:   Mon Dec 9 13:45:43 2013 -0600

    usb: core: allow isoc URBs for wireless devices with an interval < 6

    In usb_submit_urb, do not fail if an isoc URB for a wireless USB device
    has an interval < 6.  Per WUSB spec, isoc endpoints can support values
    from 1-16.  Valid values for interrupt URBs for wireless USB devices are
    still 6-16.

    Signed-off-by: Thomas Pugliese <thomas.pugliese@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/urb.c b/drivers/usb/core/urb.c
index e622083..07c58af 100644
--- a/drivers/usb/core/urb.c
+++ b/drivers/usb/core/urb.c
@@ -492,9 +492,9 @@ int usb_submit_urb(struct urb *urb, gfp_t mem_flags)
   /* too small? */
   switch (dev->speed) {
   case USB_SPEED_WIRELESS:
-   if (urb->interval < 6)
+   if ((urb->interval < 6)
+    && (xfertype == USB_ENDPOINT_XFER_INT))
     return -EINVAL;
-   break;
   default:
    if (urb->interval <= 0)
     return -EINVAL;
-- 
1.7.1